flocx_market/db/sqlalchemy/api.py

```diff
@@ -66,7 +66,8 @@ def offer_get_all_unexpired(context):
 
 def offer_create(values, context):
     values['marketplace_offer_id'] = uuidutils.generate_uuid()
-    values['project_id'] = context.project_id
+    if 'project_id' not in values:
+        values['project_id'] = context.project_id
```

Comment:
Should we throw an error (or at least log a warning) if `values` has no `project_id` key?

Reply:
Oslo.db. will throw the error for you, as the project_id can't be null.

Incidentally this is a bit of a compromise; many services which expect objects only to be created by users through some UI will always use context.project_id here. However offers are actually always created through the provider, which uses a single set of credentials from their config file.

It's possible that we don't want this context.project_id line at all, but I thought that maybe there's some corner case where somehow an offer is created outside of the provider or something... ?

Reply:
That's what I meant about throwing an error: I don't think we ever want to default to `context.project_id` here. At least, I'm trying to think of a situation where that would make sense and I'm not coming up with one.

Reply:
That's fair! I can take this bit out.

Reply:
Okay, should be taken out now!